composeApp/src/commonMain/kotlin/net/adhikary/mrtbuddy/ui/screens/transactionlist/TransactionListViewModel.kt

```
@@ -150,4 +154,36 @@ class TransactionListViewModel(
+                }
+
+                val csvContent = CsvExportService.generateCsv(allTransactions, cardIdm)
+                val filename = CsvExportService.generateFilename(
+                    state.value.cardName,
+                    Clock.System.now().toEpochMilliseconds()
+                )
+
+                fileSharer.share(csvContent, filename, "text/csv")
+
+                _state.update { it.copy(isExporting = false) }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)